darwin/cli_functions.py

```diff
@@ -830,22 +832,23 @@ def dataset_convert(dataset_slug: str, format: str, output_dir: Optional[PathLik
         The folder where the exported annotation files will be. If None it will be the inside the
         annotations folder of the dataset under 'other_formats/{format}'. The Defaults to None.
     """
-    client: Client = _load_client()
+    identifier: DatasetIdentifier = DatasetIdentifier.parse(dataset_slug)
+    client: Client = _load_client(team_slug=identifier.team_slug)
     exporter_module = import_module(f"darwin.exporter.formats.{format}")
     try:
         parser: ExportParser = getattr(exporter_module, "export")
     except AttributeError:
         _error(f"Unsupported import format, currently supported: {export_formats}")
 
     try:
-        dataset: RemoteDataset = client.get_remote_dataset(dataset_identifier=dataset_slug)
+        dataset: RemoteDataset = client.get_remote_dataset(dataset_identifier=identifier.dataset_slug)
```

Comment:
Might be out of context, but here the api is confusing. It looks like `get_remote_dataset` needs `dataset_identifier` (which one would assume to be of `DatasetIdentifier` type). Instead `dataset_identifier` is actually a dataset slug. Can we change the nomenclature here?

(Disclaimer: I'm aware we're potentially breaking the API, but consistency is in my opinion more important in v0.x.x)

An alternative could be accepting either a `dataset_identifier: DatasetIdentifier` or a `dataset_slug: str`.